.werft/jobs/build/deploy-to-preview-environment.ts

```diff
@@ -454,16 +447,8 @@ async function deployToDevWithInstaller(werft: Werft, jobConfig: JobConfig, depl
         exec(`werft log result -d "dev installation" -c github-check-preview-env url ${url}/workspaces`);
     }
 
-    try {
-        werft.log(installerSlices.DEPLOYMENT_WAITING, "Server not ready. Let the waiting...commence!");
-        exec(`kubectl -n ${namespace} rollout status deployment/server --timeout=10m`, { slice: installerSlices.DEPLOYMENT_WAITING });
-        werft.done(installerSlices.DEPLOYMENT_WAITING);
-    } catch (err) {
-        if (!jobConfig.mainBuild) {
-            werft.fail(installerSlices.DEPLOYMENT_WAITING, err);
-        }
-        exec('exit 0')
-    }
+    await waitUntilAllPodsAreReady(deploymentConfig.namespace, { slice: installerSlices.DEPLOYMENT_WAITING })
```

Comment:
Maybe you're missing a `werft.log(vmSlices.DEPLOYMENT_WAITING, 'Waiting for deployment')` or something to that effect here.

In any case, Werft is displaying the phases/logs correctly, so could be that the instrumentation in the Werft class isn't following excatly the semantics that Werft has for when to close phases.

Reply:
I've added that line.